tests/conftest.py

```diff
@@ -1568,18 +1568,22 @@ def collect_db_dump(request, duthosts):
 @pytest.fixture(scope="module", autouse=True)
 def verify_new_core_dumps(duthost):
     if "20191130" in duthost.os_version:
-        pre_existing_cores = duthost.shell('ls /var/core/ | grep -v python | wc -l')['stdout']
+        pre_existing_cores = duthost.shell('ls /var/core/ | grep -v python')['stdout'].split()
     else:
-        pre_existing_cores = duthost.shell('ls /var/core/ | wc -l')['stdout']
-    
+        pre_existing_cores = duthost.shell('ls /var/core/')['stdout'].split()
+
     yield
     if "20191130" in duthost.os_version:
-        coredumps_count = duthost.shell('ls /var/core/ | grep -v python | wc -l')['stdout']
+        cur_cores = duthost.shell('ls /var/core/ | grep -v python')['stdout'].split()
     else:
-        coredumps_count = duthost.shell('ls /var/core/ | wc -l')['stdout']
-    if int(coredumps_count) > int(pre_existing_cores):
-        pytest.fail("Core dumps found. Expected: {} Found: {}. Test failed".format(pre_existing_cores,\
-            coredumps_count))
+        cur_cores = duthost.shell('ls /var/core/')['stdout'].split()
+
+    found_dumps = len(cur_cores) - len(pre_existing_cores)
+    new_core_dumps = [x for x in cur_cores if x not in pre_existing_cores]
```

Comment:
Is it better to move this line in if condition? Because if `found_dumps` is 0, we don't have do these for loops.
And you can use `set` to find what's the differences between pre core files and current core files.
Such as:
`new_core_dumps = set(cur_cores) - set(pre_existing_cores)`

Reply:
Yes, I think that would be better